src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,125 @@
-import React from 'react';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import classNames from 'classnames';
+import { Comment, CommentData } from '../types/Comment';
+import { createComment } from '../api/comments';
+import { warningTimer } from '../utils/warningTimer';
+
+type Props = {
+  postId: number | undefined;
+  setComments: Dispatch<SetStateAction<Comment[]>>
+  setIsHasError: Dispatch<SetStateAction<boolean>>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  setComments,
+  setIsHasError,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const DefaultError = {
+    errorName: false,
+    errorEmail: false,
+    errorBody: false,
+  };
+  const nameError = 'errorName';
+  const emailError = 'errorEmail';
+  const bodyError = 'errorBody';
+  const [isError, setIsError] = useState(DefaultError);
+  const { errorName, errorBody, errorEmail } = isError;
+
+  const DefaultNewComment = {
+    postId: postId || 0,
+    name: '',
+    email: '',
+    body: '',
+  };
+  const [newComment, setNewComment] = useState<CommentData>(DefaultNewComment);
+
+  const { name: commentName, email, body } = newComment;
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const { name, value } = e.target;
+
+    if (commentName) {
+      setIsError({
```

Comment:
use callback to get access to the current state
```suggestion
      setIsError((currentErrors) => {
```